File: app/src/main/java/com/shveed/cookmegood/data/CategoryApi.java
package com.shveed.cookmegood.data;

import com.shveed.cookmegood.entity.Category;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;

public interface CategoryApi {

    String API_PATH = "http://localhost:8080";

    @GET(API_PATH + "/categories")
    Call<List<Category>> getAllCategories();

    @GET(API_PATH + "/categories/{id}")
    Call<Category> getCategoryByID();
}

File: app/src/main/java/com/shveed/cookmegood/data/CategoryApi.kt
package com.shveed.cookmegood.data

import com.shveed.cookmegood.data.dto.Category
import retrofit2.Call
import retrofit2.http.GET

interface CategoryApi {
    @get:GET("$API_PATH/categories/all")
    val allCategories: Call<List<Category?>?>?

    companion object {
       const val API_PATH = "http://10.0.2.2:8080"
    }
}